pkg/router/controller/router_controller.go

```diff
@@ -227,6 +228,70 @@ func (c *RouterController) HandleEndpoints(eventType watch.EventType, obj interf
 	c.Commit()
 }
 
+// HandleEndpointSlice handles a single EndpointSlice event and refreshes the router backend.
+func (c *RouterController) HandleEndpointSlice(eventType watch.EventType, objMeta metav1.ObjectMeta, items []discoveryv1beta1.EndpointSlice) {
+	c.lock.Lock()
+	defer c.lock.Unlock()
+
+	ipAddress := func(addressType discoveryv1beta1.AddressType, address string) string {
```

Comment:
Does this func need to be inline? (Apologies if I have already asked you this)

Reply:
I deliberately scoped this to the function - I don't know if this has general applicability elsewhere.